pkg/templates/templates.go

```diff
@@ -368,7 +368,7 @@ func UsesEncryption(template []byte, startDelim string, stopDelim string) bool {
 	// {{ ... | protect }}
 	d1 := regexp.QuoteMeta(startDelim)
 	d2 := regexp.QuoteMeta(stopDelim)
-	re := regexp.MustCompile(d1 + `(\s*fromSecret\s+.*|\s*copySecretData\s+.*|.*\|\s*protect\s*)` + d2)
+	re := regexp.MustCompile(d1 + `-?(\s*fromSecret\s+.*|\s*copySecretData\s+.*|.*\|\s*protect\s*)-?` + d2)
```

Comment:
Can you add tests for this to `TestUsesEncryption`?

Separately, outside of this PR, I think we need to improve this regex. It doesn't cover things like `{{ printf "secret: %q" (fromSecret "foo" "bar" "baz") }}`